Mathlib/CategoryTheory/Limits/Indization/IndObject.lean

```diff
@@ -0,0 +1,110 @@
+/-
+Copyright (c) 2024 Markus Himmel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Markus Himmel
+-/
+import Mathlib.CategoryTheory.Limits.Presheaf
+
+/-!
+# Ind-objects
+
+For a presheaf `A : Cᵒᵖ ⥤ Type v` we define the type `IndObjectPresentation A` of presentations
+of `A` as a small filtered colimit of representable presheaves and define the predicate
+`IsIndObject A` asserting that there is at least one such presentation.
+
+## Future work
+
+A presheaf is an ind-object if and only if the category `CostructuredArrow yoneda A` is filtered
+and finally small. In this way, `CostructuredArrow yoneda A` can be thought of the universal
+indexing category for the representation of `A` as a small filtered colimit of representable
+presheaves.
+
+There are various useful ways to understand natural transformations between ind-objects in terms
+of their presentations.
+
+The ind-objects form a locally `v`-small category `IndCategory C` which has numerous interesting
+properties.
+
+## Implementation notes
+
+One might be tempted to introduce another universe parameter and consider being a `w`-ind-object
+as a property of presheaves `C ⥤ TypeMax.{v, w}`. This comes with significant technical hurdles.
+The recommended alternative is to consider ind-objects over `ULiftHom.{w} C` instead.
+
+## References
+* [M. Kashiwara, P. Schapira, *Categories and Sheaves*][Kashiwara2006], Chapter 6
+-/
+
+universe v u
+
+namespace CategoryTheory.Limits
+
+variable {C : Type u} [Category.{v} C]
+
+/-- The data that witnesses that a presheaf `A` is an ind-object. It consists of a small
+    filtered indexing category `I`, a diagram `F : I ⥤ C` and the data for a colimit cocone on
+    `F ⋙ yoneda : I ⥤ Cᵒᵖ ⥤ Type v` with cocone point `A`. -/
+structure IndObjectPresentation (A : Cᵒᵖ ⥤ Type v) where
+  /-- The indexing category of the filtered colimit presentation -/
+  (I : Type v)
```

Comment:
This syntax with braces is rarely used in mathlib4; it appears mostly only when the structure has several fields of the same type:
```suggestion
  I : Type v
```
(And similarly below.)

Reply:
Thanks, I don't know why I keep repeating this mistake.